__tests__/components/App.test.js

```diff
@@ -4,14 +4,18 @@ import 'jsdom-global/register'
 import React from 'react'
 import { mount, shallow } from "enzyme"
 import { MemoryRouter } from "react-router"
-import RootContainer from '../../src/components/RootContainer'
+import { Provider } from 'react-redux'
 import App from '../../src/components/App'
-import Config from '../../src/Config'
+import LoginPanel from '../../src/components/LoginPanel'
 import HomePage from '../../src/components/HomePage'
 import Editor from '../../src/components/editor/Editor'
 import Browse from '../../src/components/editor/Browse'
+import CanvasMenu from '../../src/components/CanvasMenu'
 import Footer from '../../src/components/Footer'
-import ImportResourceTemplate from "../../src/components/editor/ImportResourceTemplate";
+import ImportResourceTemplate from '../../src/components/editor/ImportResourceTemplate';
+
+jest.mock('../../src/components/editor/Editor')
```

Comment:
wasn't clear to me what exactly, but seems like `Editor` component wants some props that our test mounted `App` doesn't provide, and if it doesn't get them, the `/templates` route gets rendered.  so just mock the component implementation since `Editor`'s tests should confirm that it works correctly, and we can just make these `App` unit tests about whether `App`'s routing is set up as expected.

if we want to make sure `App` and `Editor` are wired up correctly, i'm guessing that it'd be both easier and more confidence inspiring to write a fuller integration test using puppeteer.

Reply:
Agree that this makes sense, and that testing integration between these components best fits an integration test than a unit test.